### PR TITLE
fix(completion): `super::` completion at crate root and module depth aware

### LINKED ISSUE #12439: `use` erroneously suggests `super::` as a completion option even when it's not valid.
Consider you're in the `main.rs` of a crate and type:

```rust
use $0
```

Right now, this will offer `super::` as a completion, when there is *no* super. 

Additionally, following will offer a `super::` completion even when not relevant:

```rust
use super::$0;
use super::super::$0;
use super::super::super::$0;
// ... ad-infinitum...
```

---

(as an aside, I'm mainly opening these in response to the zulip thread asking for "minor inconveniences" in r-a's completion, hope y'all don't mind the issue spam!)

### PR DESCRIPTION
**Pull request by feniljain**
- should close #12439
- Suggest super according to the depth in tree

### CODE REVIEW

File: crates/ide-completion/src/context/analysis.rs [hunk 1]
```diff
@@ -899,10 +899,25 @@ impl<'a> CompletionContext<'a> {
...
+
+                        // For understanding how and why super_chain_len is calculated the way it
+                        // is check the documentation at it's definition
+                        let mut segment_count = 0;
+                        let super_count = iter::successors(Some(path.clone()), |p| p.qualifier())
+                            .take_while(|p| {
+                                p.segment()
+                                    .and_then(|s| {
+                                        segment_count += 1;
+                                        return s.super_token();
```
Comment (Veykril) [inline crates/ide-completion/src/context/analysis.rs:911]:
```suggestion
                                        s.super_token()
```

**Review by Veykril — COMMENTED** [review]
Small nit, otherwise looks good!

**Comment by feniljain** [comment]
Ah I guess edited comments don't work so trying one more time
@bors r+

**Pull request merged**